xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -485,7 +485,7 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
 
       // grab the next character
       Character* ch = &characters.front();
-      if (ch->m_letter == 0)
+      if (ch->m_glyphAndStyle == 0)
```

Comment:
Yep, will change.

Reply:
Will not change. :-) This line is not to detect any real character. It is something hacky (not invented ny me) to detect an empty entry pushed to the vector in line 461 (https://github.com/xbmc/xbmc/pull/21708/files#diff-13eb64264879fab58d04f4b2504916173159f92a09b714005363f649ca813cfeL461) if GetCharacter returned nullptr.

Reply:
It cannot be nullptr, because the vector holds instances, not pointers. 

And I even tried the other day to switch from instances to pointers, but that leads to wrong characters or even crashes, because the characters are obtained from `GetCharacter` and those pointers must not be "cached" across several `GetCharacter` calls because the next call to `GetCharacter` can invalidate the pointer returned by the previous call as you know. So, copies of the characters must be kept in the `characters` vector. A big mess if you ask me...